src/NServiceBus.Transport.SQS/Configure/SqsTransport.cs

```diff
@@ -142,6 +142,53 @@ public Func<Type, string, string> TopicNameGenerator
         /// for characters</see> <see href="https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html">SQS requires</see> the payload is base64 encoded automatically.</remarks>
         public bool DoNotWrapOutgoingMessages { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum duration within which the message visibility will be renewed automatically. This
+        /// value should be greater than the longest message visibility duration specified either on the queue or on the receive request controlled by <see name="VisibilityTimeout"/>.
+        /// </summary>
+        /// <value>The maximum duration during which message visibility are automatically renewed. The default value is 5 minutes. The renewal can be disabled by passing <see cref="TimeSpan.Zero"/>.</value>
+        public TimeSpan MaxAutoMessageVisibilityRenewalDuration
+        {
+            get => maxAutoMessageVisibilityRenewalDuration;
+            set
+            {
+                var visibilityTimeoutInSeconds = (int)value.TotalSeconds;
+                ArgumentOutOfRangeException.ThrowIfNegative(visibilityTimeoutInSeconds, nameof(MaxAutoMessageVisibilityRenewalDuration));
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(visibilityTimeoutInSeconds, TimeSpan.FromHours(12).TotalSeconds, nameof(MaxAutoMessageVisibilityRenewalDuration));
+
+                maxAutoMessageVisibilityRenewalDuration = value;
+            }
+        }
+
+        TimeSpan maxAutoMessageVisibilityRenewalDuration = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Gets or sets the message visibility timeout for the receive request. This value overrides the queue visibility timeout
+        /// </summary>
+        /// <value>The default value is <c>null</c></value>
+        public TimeSpan? MessageVisibilityTimeout
+        {
+            get => messageVisibilityTimeout;
+            set
+            {
+                messageVisibilityTimeout = value;
+                if (!value.HasValue)
+                {
+                    return;
+                }
```

Comment:
If `value` is null, `messageVisibilityTimeout` will be null. So far, so good. Why the `if` is not setting to null also the `messageVisibilityTimeoutInSeconds`?

```suggestion
                messageVisibilityTimeout = value;
                if (!value.HasValue)
                {
                    messageVisibilityTimeoutInSeconds = null;
                    return;
                }
```